string timestamp
string phrase
string action_id
KeyValue[] params
string user_id
